## Strukturierte Datensätze :
### Strukturierte Daten

Bei <a href="https://en.wikipedia.org/wiki/Data_science">Data Science</a> geht es um die Gewinnung von Wissen aus 
<a href="https://en.wikipedia.org/wiki/Data">Daten</a>
. Daten sind eine spezifische Form von <a href="https://en.wikipedia.org/wiki/Information">Informationen</a> und weisen verschiedene Abstraktions- und Strukturniveaus auf (<a href="https://en.wikipedia.org/wiki/Data_model">strukturiert</a>, <a href="https://en.wikipedia.org/wiki/Semi-structured_data">halbstrukturiert</a> oder <a href="https://en.wikipedia.org/wiki/Unstructured_data">unstrukturiert</a>). <br> Eine sehr verbreitete <a href="https://en.wikipedia.org/wiki/Data_structure">Datenstruktur</a> ist ein Array. In verschiedenen Bereichen gibt es andere Bezeichnungen für einen solchen Datentyp, die synonym verwendet werden, z. B. <a href="https://en.wikipedia.org/wiki/Matrix_(mathematics)">Matrix</a> in der Mathematik, <a href="https://en.wikipedia.org/wiki/Table_(database)">Tabelle</a> in Datenbanken, <a href="https://en.wikipedia.org/wiki/SpreadsheetDatenrahmen">Tabellenkalkulation</a> und #<a href="https://www.geeksforgeeks.org/python-pandas-dataframe/">data.frame</a> , der eine grundlegende Python-Objektklasse ist zB : (Pandas DataFrame)#. <br> Daten eines solchen Typs bestehen aus Beobachtungen und entsprechenden Variablen, die oft als Merkmale bezeichnet werden.

<head>
<title>Tabelle mit Breitenangaben in Prozent</title>
    <body>
<head>
<title>Tabelle mit Breitenangabe in Pixeln</title>
</head>

<body>

<table border="1" width="500" height="200">
  <tr>
    <td width="300" height="100">id</td>
    <td width="300" height="100">Name</td>
    <td width="300" height="100">Age</td>
  </tr>
  <tr>
    <td>1</td>
    <td>John</td>
    <td>26</td>
  </tr>
  <tr>
    <td>2</td>
    <td>Alice</td>
    <td>20</td>
   </tr>
  <tr>
    <td>3</td>
    <td>Mike</td>
    <td>21</td>
   <tr>
    <td>4</td>
    <td>Anne</td>
    <td>25</td>
  </tr>
</table>

</body>
</html>

In diesem Beispiel entsprechen die **Beobachtungen** (*Stichprobe* genannt) einer Anzahl von Personen. Jede beobachtete Person wird durch eine Reihe von **Variablen** (so genannte *Merkmale*) charakterisiert: durch eine Identifikationsnummer (id), durch einen Namen und durch ein Alter. In unserem Beispiel ist es sehr einfach, sich durch einen Blick auf die Tabelle einen Gesamteindruck von den Daten selbst zu verschaffen. Wir erkennen sofort, dass es in unserer Stichprobe 4 Personen gibt, zwei Frauen und zwei Männer. Außerdem sehen wir sofort, dass die jüngste Person 20 Jahre alt ist und Alice heißt und die älteste Person 26 Jahre alt ist und John heißt; perfekt!

Anwendungen in der realen Welt enthalten jedoch oft eine große Menge an Daten. Hunderte, Tausende, Millionen oder sogar Milliarden von Beobachtungen, kombiniert mit Tausenden von Variablen, können einen Datensatz bilden. Für den Menschen ist es unmöglich, allein durch die Betrachtung solcher Datensätze irgendwelche Schlussfolgerungen über die Daten zu ziehen. Daher reduzieren wir die Daten auf eine überschaubare Größe, indem wir Tabellen erstellen, Diagramme zeichnen oder zusammenfassende Maße wie Durchschnittswerte berechnen. Diese Art von statistischen Methoden wird als **deskriptive Statistik** bezeichnet (Mann 2012).

### Der Studenten Datensatz
In diesem Abschnitt werden wir einen Datensatz namens Studenten untersuchen. Sie können die Datei $students.csv$ <a href="https://userpage.fu-berlin.de/soga/200/2010_data_sets/students.csv">hier</a>  herunterladen. Zunächst laden wir den Datensatz, geben ihm einen geeigneten Namen und verschaffen uns einen Eindruck von seiner Struktur und Größe, indem wir die Funktion $print()$ für den Datensatz aufrufen.

In [6]:
import pandas as pd
studentsdb=pd.read_csv('students2.csv')
print(studentsdb)

      stud.id                 name  gender  age  height  weight    religion  \
1      833917  Gonzales, Christina  Female   19     160    64.8      Muslim   
2      898539       Lozano, T'Hani  Female   19     172    73.0       Other   
3      379678       Williams, Hanh  Female   22     168    70.6  Protestant   
4      807564          Nem, Denzel    Male   19     183    79.7       Other   
5      383291      Powell, Heather  Female   21     175    71.4    Catholic   
...       ...                  ...     ...  ...     ...     ...         ...   
8235   818044          Her, Joshua    Male   22     181    79.1  Protestant   
8236   560166           Reid, Jose    Male   23     178    72.6    Catholic   
8237   380174     Villa, Raechelle  Female   20     169    67.7       Other   
8238   540967         Ngo, Preston    Male   23     195    98.6       Other   
8239   413207     Bernhard, Estela  Female   19     170    69.0      Muslim   

      nc.score semester                   major    

Der Studentendatensatz besteht aus 8239 Zeilen, von denen jede einen bestimmten Studenten repräsentiert, und 16 Spalten, von denen jede einer Variable/einem Merkmal entspricht, das sich auf diesen bestimmten Studenten bezieht. Diese selbsterklärenden Variablen sind: stud.id, name, gender, age, height, weight, religion, nc.score, semester, major, minor, score1, score2, online.tutorial, graduated, salary. Neben dem jeweiligen Variablennamen listet die Funktion $type()$ die $Klasse$ jeder einzelnen Variablen auf. Alle Objekte in Python haben eine Klasse, z. B. $numerische$ Datentypen , die in die Unterklassen $(int)$ Ganzzahlen , $(float)$ Gleitkommazahlen und $(imag)$komplexe Zahlen, eingeteilt wird, neben anderen.

In den nächsten Abschnitten werden wir die deskriptiven Statistiken des *Studentendatensatzes* genauer untersuchen.
